types/node/index.d.ts

```diff
@@ -772,23 +773,34 @@ declare module "http" {
     import * as stream from "stream";
     import { URL } from "url";
 
-    export interface RequestOptions {
+    export interface HttpHeaders {
```

Comment:
Please note that incoming headers will never contain numbers but setting response headers allows nubmers (as they are converted internally to strings). Therefore using HttpHeaders in incoming message forces everyone to check if type is string or number.
I think we should keep two distinct types for incoming and outgoing headers.